rust/src/common.rs

```diff
@@ -10,3 +13,29 @@ macro_rules! take_until_and_consume (
     }
   );
 );
+
+/// Convert a String to C-compatible string
+///
+/// This function will consume the provided data and use the underlying bytes to construct a new
+/// string, ensuring that there is a trailing 0 byte. This trailing 0 byte will be appended by this
+/// function; the provided data should *not* contain any 0 bytes in it.
+///
+/// Returns a valid pointer, or NULL
+pub fn rust_string_to_c(s: String) -> *mut c_char {
```

Comment:
can this be in its own commit, with https://github.com/OISF/suricata/pull/4663/commits/9e58350b760c1bf450b7b9cfd6c2922d28953883 squashed into it?

Reply:
Done